### PR TITLE
✅ add `yarn test:unit:watch` script

### PR DESCRIPTION
**Pull request by BenoitZugmeyer**
## Motivation

Small trick that I always use when writing unit tests

## Changes

Add a `yarn test:unit:watch` script

## Testing

---

I have gone over the [contributing](https://github.com/DataDog/browser-sdk/blob/main/CONTRIBUTING.md) documentation.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -20,6 +20,7 @@
     "publish:npm": "TARGET_DATACENTER=us BUILD_MODE=release yarn build && lerna publish from-package",
     "test": "yarn test:unit",
     "test:unit": "karma start ./test/unit/karma.local.conf.js",
+    "test:unit:watch": "yarn test:unit --no-single-run",
```
Comment (bcaudan) [inline package.json:23]: I also wanted to open a PR for something similar from some time.
What would you think of having this behavior for `yarn test` directly?

Comment (BenoitZugmeyer) [reply package.json:23]: It is somewhat unusual to have this as the default test command, so as a newcomer I wouldn't expect this behavior. But if it is discoverable as a separate test as well, I wouldn't mind :)

Comment (bcaudan) [reply package.json:23]: So, keep `test:unit:watch` and use it from `test`?

**Review by bcaudan — APPROVED** [review]
Sweet!

**Pull request merged**